contributors/design-proposals/federated-statefulsets.md

```diff
@@ -0,0 +1,491 @@
+ goes down, the quorum in that cluster is dead, but the quora in all the other clusters remain
+ up accessible locally to the in-cluster apps. (Similar to use case 1 above).
+
+2 - A federated etcd statefulset of 3 or more replicas across 2 or more then 2 clusters
+which creates a global quorum able to withstand the temporary failure of any one cluster,
+as well as the permanent failure of any node in any cluster. (Similar to use case 2 above).
+
+3 - A federated cassandra db statefulset of 100 replicas across, for example, 5 clusters.
+(The instances can have multiple identities each; apps and peer instances can choose to
+preferentially connect to local dns identities, but can also connect using global dns identity,
+if need be, for example to connect to peers in other federated clusters).
+
+4 - A federated statefulset of an app, which is designed to inherently support a geo-distributed
+app cluster, with replicas in mutiple federated k8s clusters. (for example galera db cluster).
+
+# Design Requirements
+
+If we consider the use cases listed above, the main design requirements can roughly be listed as:
+
+1 – An unique, consistent and discoverable identity for each replica/instance across the
+federated clusters.
+
+2 – It must be possible for the federated statefulset to first safely form an initial quorum
+(probably using cluster local replicas and identities).
+It should further also be possible to add members (probably from other clusters and global
+identities) to this initial quorum.
+
+3 – The ability to scale, across clusters in some deterministic fashion.
+
+4 – An extended design requirement (which might not be a necessary requirement for this
+phase for work) can be an ability to migrate a pod from one node (in one cluster) to
+another node (possibly in another cluster), with stable network identity along with its attached volumes.
+
+# Design Proposals
+
+The Statefulset object and APIs are already defined in k8s.
+The same needs to be exposed from federation to ensure that the interfaces remain consistent
+with a stand-alone k8s cluster.
+A separate controller at federation would however be useful.
+
+We considered two alternative designs for this proposal, both of which are further described
+in this document. The detailed design for the primary alternative is elaborated in the next section.
+A design alternative is also evaluated and listed later in the doc.
+
+## Detailed Design (Primary Alternative)
+
+In this approach, the federated statefulset controller will partly behave quite similar to federated
+replicaset or the federated deployment controller.
+The federated controller would create and monitor individual statefulsets partitioning and distributing
+the total stateful replicas across the federated clusters.
+
+As an additional proposal in this design we suggest the possibility of the pods having multiple identities.
+This facilitates application pods being able to participate in different quora, and leaves a
```

Comment:
I have given some details In reply to one your comments (https://github.com/kubernetes/community/pull/437#discussion_r124242716)
Additionally, this certainly is not something which will force a pod to have multiple identities. As mentioned the app pod "can" if apps need be. 
The two identities however are not exclusive, but orthogonal (as in our earlier discussions and per this proposal). The app can have local identity alone or local + global if it needs so. The app which needs to use "only" global identity needs to ignore the local one.